drivers/snmp-ups.c

```diff
@@ -914,12 +977,40 @@ void su_status_set(snmp_info_t *su_info_p, long value)
 void su_alarm_set(snmp_info_t *su_info_p, long value)
 {
 	const char *info_value = NULL;
+	char alarm_info_value[SU_LARGEBUF];
+	/* number of the outlet or phase */
+	int item_number = -1;
 
-	upsdebugx(2, "SNMP UPS driver : entering su_alarm_set()");
+	upsdebugx(2, "SNMP UPS driver : entering su_alarm_set(%s)", su_info_p->info_type);
 
 	if ((info_value = su_find_infoval(su_info_p->oid2info, value)) != NULL)
 	{
 		if (strcmp(info_value, "")) {
+			/* Special handling for outlet & outlet groups alarms */
+			if ((su_info_p->flags & SU_OUTLET)
+				|| (su_info_p->flags & SU_OUTLET_GROUP)) {
+				/* Extract template number */
+				item_number = extract_template_number(su_info_p->flags, su_info_p->info_type);
+
+				/* Inject in the alarm string */
+				sprintf(alarm_info_value, info_value, 
+					(su_info_p->flags & SU_OUTLET_GROUP)?"outlet group ":"outlet ",
+					item_number);
+				info_value = &alarm_info_value[0];
+			}
+			/* Special handling for phase alarms
+			 * Note that SU_*PHASE flags are cleared, so match the 'Lx'
+			 * start of path */
+			if (su_info_p->info_type[0] == 'L') {
+				/* Extract phase number */
+				item_number = atoi(strchr(su_info_p->info_type, 'L')+1);
```

Comment:
Efficiency vs. readability: fromt the "if" above, we know that "su_info_p->info_type[0] == 'L'", so for `atoi` we can pass something like "su_info_p->info_type+1"?